pkg/reconciler/v1alpha1/clusteringress/resources/secret.go

```diff
@@ -0,0 +1,95 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/knative/serving/pkg/apis/networking"
+	"github.com/knative/serving/pkg/apis/networking/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+)
+
+// GetSecrets gets the all of the secrets referenced by the given ClusterIngress, and
+// returns a map whose key is the a secret namespace/name key and value is pointer of the secret.
+func GetSecrets(ci *v1alpha1.ClusterIngress, secretLister corev1listers.SecretLister) (map[string]*corev1.Secret, error) {
+	secrets := map[string]*corev1.Secret{}
+	for _, tls := range ci.Spec.TLS {
+		ref := fmt.Sprintf("%s/%s", tls.SecretNamespace, tls.SecretName)
+		if _, ok := secrets[fmt.Sprintf("%s/%s", tls.SecretNamespace, tls.SecretName)]; ok {
```

Comment:
this is defined as var on the previous line.

Reply:
Done